pkg/deploy/deploy.go

```diff
@@ -206,127 +205,28 @@ func GetResources(resMap resmap.ResMap) ([]*unstructured.Unstructured, error) {
-		var componentCounter []string
-		if resourceLabels != nil {
-			for i := range resourceLabels {
-				if strings.Contains(i, labels.ODHAppPrefix) {
-					compFound := strings.Split(i, "/")[1]
-					componentCounter = append(componentCounter, compFound)
-				}
-			}
-			// Shared resource , do not delete. Remove label from disabled component
-			if len(componentCounter) > 1 || (len(componentCounter) == 1 && componentCounter[0] != componentName) {
-				found.SetLabels(resourceLabels)
-				// return, do not delete the shared resource
-				return nil
-			}
-
-			// Do not delete CRDs, as those can be used by non-odh components
-			if found.GetKind() == "CustomResourceDefinition" {
-				return nil
-			}
-		}
-
-		existingOwnerReferences := obj.GetOwnerReferences()
-		selector := labels.ODH.Component(componentName)
-		// only removed the resource with our label applied, not the same name resource maually created by user
-		if existingOwnerReferences == nil && resourceLabels[selector] == "true" {
-			return cli.Delete(ctx, found)
-		}
-
-		found.SetOwnerReferences([]metav1.OwnerReference{})
-		data, err := json.Marshal(found)
-		if err != nil {
-			return err
-		}
-
-		err = cli.Patch(ctx, found, client.RawPatch(types.ApplyPatchType, data), client.ForceOwnership, client.FieldOwner(owner.GetName()))
-		if err != nil {
-			return err
-		}
-
-		return cli.Delete(ctx, found)
-	}
-
-	// Create the resource if it doesn't exist and component is enabled
-	if apierrs.IsNotFound(err) {
-		// Set the owner reference for garbage collection
-		// Skip set on CRD, e.g. we should not delete notebook CRD if we delete DSC instance
-		// Skip on OdhDashboardConfig CR, because we want user to be able to update it
-		if found.GetKind() != "CustomResourceDefinition" && found.GetKind() != "OdhDashboardConfig" {
-			if err = ctrl.SetControllerReference(owner, metav1.Object(obj), cli.Scheme()); err != nil {
-				return err
-			}
-		}
-
-		return cli.Create(ctx, obj)
-	}
-
-	// Exception: ODHDashboardConfig should not be updated even with upgrades
-	// TODO: Move this out when we have dashboard-controller
-	if found.GetKind() == "OdhDashboardConfig" {
-		// Do nothing, return
-		return nil
-	}
-
-	// Kserve specific workflow:
-	// TODO: Remove this when we have generalize custom config requirements across all components
-	if componentName == "kserve" {
-		// do not reconcile kserve resource with annotation "opendatahub.io/managed: false"
-		if found.GetAnnotations()[annotation.ManagedByODHOperator] == "false" {
-			return nil
-		}
-		// do not patch resources field in Kserve deployment i.e allows users to update resources field
-		if err := removeResourcesFromDeployment(obj); err != nil {
-			return err
-		}
-	}
-
-	// JIRA 6889.  odh-model-controller will be covered by either kserve or model-mesh case
-	if componentName == "model-mesh" {
-		if err := removeResourcesFromDeployment(obj); err != nil {
-			return err
-		}
+	// Return if error getting resource in cluster
```

Comment:
The comment is a bit misleading now since getResource does not return error if it was NotFound.

I would say the original logic flow is more clear to me. If getResource() returns (nil, error) if there was error including NotFound and returns (found, nil) if it could get the object, than original check here `if IgnoreNotFound(err) != nil { return err }` will catch all the errors and we can handle NotFound with `found` to nil check.

Then getResource() as 
```
func getResource(ctx context.Context, cli client.Client, obj *unstructured.Unstructured) (*unstructured.Unstructured, error) {
	found := &unstructured.Unstructured{}
	// Setting gvk is required to do Get request
	found.SetGroupVersionKind(obj.GroupVersionKind())
	err := cli.Get(ctx, types.NamespacedName{Name: obj.GetName(), Namespace: obj.GetNamespace()}, found)
	if err != nil {
		return nil, err
	}
	return found, nil
}
```
will not break idiomatic way of propagating error.

But I'll not insist.

Reply:
Yeah I will update this. Thanks!

Reply:
Updated